packages/lexical-playground/src/plugins/TableCellResizer/index.tsx

```diff
@@ -74,20 +76,48 @@ function TableCellResizer({editor}: {editor: LexicalEditor}): JSX.Element {
   };
 
   useEffect(() => {
-    return editor.registerNodeTransform(TableNode, (tableNode) => {
-      if (tableNode.getColWidths()) {
-        return tableNode;
-      }
+    // Track table existence using mutation listener
+    const unregisterMutation = editor.registerMutationListener(
+      TableNode,
+      (nodeMutations) => {
+        for (const [nodeKey, mutation] of nodeMutations) {
+          if (mutation === 'destroyed') {
+            tableKeysRef.current.delete(nodeKey);
+          } else {
+            tableKeysRef.current.add(nodeKey);
+          }
+        }
+        setHasTable(tableKeysRef.current.size > 0);
+      },
+    );
+
+    // Register table node transform
+    const unregisterTransform = editor.registerNodeTransform(
+      TableNode,
+      (tableNode) => {
+        if (tableNode.getColWidths()) {
+          return tableNode;
+        }
 
-      const numColumns = tableNode.getColumnCount();
-      const columnWidth = MIN_COLUMN_WIDTH;
+        const numColumns = tableNode.getColumnCount();
+        const columnWidth = MIN_COLUMN_WIDTH;
 
-      tableNode.setColWidths(Array(numColumns).fill(columnWidth));
-      return tableNode;
-    });
+        tableNode.setColWidths(Array(numColumns).fill(columnWidth));
+        return tableNode;
+      },
+    );
+
+    return () => {
+      unregisterMutation();
+      unregisterTransform();
+    };
```

Comment:
The normal pattern for this is to use

```suggestion
    return mergeRegsiter(
      unregisterMutation,
      unregisterTransform,
    );
```

Although typically the calls are also inline, which reduces code size a bit further and makes sure they're all actually called

```
    return mergeRegsiter(
      editor.registerMutationListener(…),
      editor.registerNodeTransform(…),
    );
```

Reply:
Thanks! I've updated the code to use `mergeRegister` for combining the cleanup functions and also made the calls inline. Will keep it in mind for future PRs.